experimental/dds/tree2/src/feature-libraries/chunked-forest/codec/nodeShape.ts

```diff
@@ -0,0 +1,179 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import {
+	FieldKey,
+	GlobalFieldKey,
+	ITreeCursorSynchronous,
+	LocalFieldKey,
+	TreeSchemaIdentifier,
+	forEachField,
+	isGlobalFieldKey,
+	keyFromSymbol,
+	symbolFromKey,
+} from "../../../core";
+import { brand, fail } from "../../../util";
+import { BufferFormat, IdentifierToken, Shape } from "./chunkEncodingGeneric";
+import { Counter, DeduplicationTable } from "./chunkCodecUtilities";
+import { EncodedChunkShape, EncodedFieldShape, EncodedValueShape } from "./format";
+import {
+	NodeEncoderShape,
+	FieldShape,
+	FieldEncoderShape,
+	EncoderCache,
+	encodeValue,
+} from "./compressedEncode";
+
+export class NodeShape extends Shape<EncodedChunkShape> implements NodeEncoderShape {
+	// TODO: Ensure uniform chunks, encoding and identifier generation sort fields the same.
+	private readonly fields: FieldShape<FieldKey>[];
+	private readonly explicitKeys: Set<FieldKey>;
+
+	public constructor(
+		public readonly type: undefined | TreeSchemaIdentifier,
+		public readonly value: EncodedValueShape,
```

Comment:
we could make fields, type, value, local, global extraLocal and extraGlobal all postfixed with Shape here. Since all fields in here describe shapes, I think its nice to leave it off.

ex: NodeShape.value being a shape kinda makes sense, and NodeShape.valueShape is kinda redundant.